packages/gatsby-plugin-typescript/src/gatsby-node.js

```diff
@@ -1,15 +1,19 @@
-const { tsPresetsFromJsPresets } = require(`./`)
+// const { tsPresetsFromJsPresets } = require(`./`)
 
 const resolvableExtensions = () => [`.ts`, `.tsx`]
 
+function onCreateBabelConfig({ actions }, pluginOptions) {
+  actions.setBabelPreset({
+    name: `@babel/preset-typescript`,
```

Comment:
To *actually* fix this we need to remove `@babel/preset-flow` from the default presets as well. I'm not sure how to do this with the changes introduced in #6801.